cordapp-upgrades/README.md

```diff
@@ -0,0 +1,158 @@
+<p align="center">
+  <img src="https://www.corda.net/wp-content/uploads/2016/11/fg005_corda_b.png" alt="Corda" width="500">
+</p>
+
+# The Upgrade CorDapp
+
+This directory contains a number of versions of the same CorDapp (based heavily on the Obligation sample CorDapp). It is
+designed to showcase ways in which a CorDapp might be upgraded such that the app remains backwards compatible between
+versions.
+
+The CorDapp is divided into contracts and workflows. These are versioned separately.
+
+# Current versions
+
+The CorDapp contract and workflow versions are described below.
+
+## Contracts
+
+### Version 1
+
+The initial contract version.
+
+### Version 2
+
+This second version adds a new field to the obligation state that allows the borrower to default on the obligation. A
+command is added to the contract to facilitate this. The new field has to be nullable in order to ensure backwards
```

Comment:
"The new field has to be nullable" (in Kotlin only?)
This is likely to puzzle end users so recommend a small explanation and adding a link to the rules of Object serialization evolution:
https://docs.corda.net/head/serialization.html?highlight=object%20serialization#custom-types